Sources/Foundation/Attributes.swift

```diff
@@ -317,3 +319,24 @@ class BooleanValue: NSObject, NSCoding {
         self.value = value
     }
 }
+
+
+// TODO: make this  private
+public struct Tag: Codable, Equatable, Hashable {
+    public static func == (lhs: Tag, rhs: Tag) -> Bool {
+        return lhs.rawValue == rhs.rawValue
+    }
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(rawValue)
+    }
+    
+    public var rawValue: String
+    public var expiresAt: Date?
+}
+
+extension Tag: ExpressibleByStringLiteral {
+    public init(stringLiteral: String) {
+        self.rawValue = stringLiteral
+    }
+}
```

Comment:
Where should this go if its internal only to ContextManager? Inside the same file?

Reply:
That's fine; if it isn't referenced by anything outside of Foundation you can remove the `public` declarations.